docs/src/main/asciidoc/bigquery.adoc

```diff
@@ -0,0 +1,91 @@
+== Google Cloud BigQuery
+
+https://cloud.google.com/bigquery[Google Cloud BigQuery] is a fully managed, petabyte scale, low cost analytics data warehouse.
+
+Spring Cloud GCP provides https://spring.io/projects/spring-integration[Spring Integration] message handlers which can be used to build pipelines that load data into BigQuery tables.
+
+Spring Cloud GCP the following message handlers:
+
+* `BigQueryFileMessageHandler` - Handle messages containing data files and loads these into BigQuery tables.
+
+Maven coordinates, using <<getting-started.adoc#_bill_of_materials, Spring Cloud GCP BOM>>:
+
+[source,xml]
+----
+<dependency>
+    <groupId>org.springframework.cloud</groupId>
+    <artifactId>spring-cloud-gcp-bigquery</artifactId>
+</dependency>
+<dependency>
+    <groupId>org.springframework.integration</groupId>
+    <artifactId>spring-integration-core</artifactId>
+</dependency>
+----
+
+Gradle coordinates:
+
+[source,subs="normal"]
+----
+dependencies {
+    compile group: 'org.springframework.cloud', name: 'spring-cloud-gcp-bigquery'
+    compile group: 'org.springframework.integration', name: 'spring-integration-core'
+}
+----
+
+=== Outbound Message Handler
+
+`BigQueryFileMessageHandler` is a Spring Integration message handler which can process files received on a https://docs.spring.io/spring-integration/docs/latest-ga/reference/html/#channel[Message Channel] and load these into a BigQuery dataset.
+This offers a convenient way to write pipelines where the goal is to load data into BigQuery.
+
+To begin using the `BigQueryFileMessageHandler`, start off by defining a `@ServiceActivator` bean for your application which returns the handler.
+
+[source,java]
+----
+@Bean
+@ServiceActivator(
+    inputChannel = "bigQueryInputChannel", outputChannel = "bigQueryOutputChannel")
+public MessageHandler messageSender(BigQuery bigQuery) {
+  return new BigQueryFileMessageHandler(bigQuery);
+}
+----
+
+There should be both an input and output channel configured for your `ServiceActivator`.
+
+* The input channel is responsible for accepting the https://docs.spring.io/spring-integration/docs/latest-ga/reference/html/#message[Messages] whose payloads contain the files you wish to load to your BigQuery table.
+* The output channel is where `BigQueryFileMessageHandler` will send all of the replies resulting from the load operations; each reply will yield a `Job` object which provides details about its corresponding load `Job`.
+
+==== Input Channel
+
+The `BigQueryFileMessageHandler` accepts messages whose payloads contain the data file being loaded.
+The accepted payload types include `byte[]`, `java.io.File`, or `java.io.InputStream`.
```

Comment:
Is payload of type `byte[]` the actual contents of the file or the file reference?